src/coreComponents/constitutive/solid/Damage.hpp

```diff
@@ -41,15 +41,24 @@
 
 #ifndef GEOSX_CONSTITUTIVE_SOLID_DAMAGE_HPP_
 #define GEOSX_CONSTITUTIVE_SOLID_DAMAGE_HPP_
-#define QUADRATIC_DISSIPATION 0
-#define LORENTZ 0
+
 #include "constitutive/solid/SolidBase.hpp"
 
 namespace geosx
 {
 namespace constitutive
 {
 
+// DAMAGE MODEL UPDATES
+//
+// NOTE: This model uses the m_newStress array to represent the stress in an
+//       elastic, "undamaged" configuration.  We then scale the results
+//       by the damage factor whenever the true stress is requested through a getter.
+//
+// TODO: This approach is probably error prone---e.g. when stress data is
+//       accessed directly through an array view.  We should refactor so
+//       the true state is saved, similar to the plasticity models
```

Comment:
To me looks like this issue is fixed now. Everything is done inside DamageUpdates::smallStrainUpdate. The undamaged stress is still computed when this function calls UPDATE_BASE::smallStrainUpdate( k, q, strainIncrement, stress, stiffness ), but in the next lines it multiplies by the damage factor. So I don't think that the undamaged stress could be wrongly called.

Previously, I used the undamaged stress to recover the strains to perform the spectral split. That was necessary because the function that computed the spectral split didn't have the strains in its scope. Now that we do everything inside the smallStrainUpdate function, the strains are easily obtained, so, we don't need to store undamaged stresses anymore.

Reply:
Okay, I've modified the TODO to just a NOTE, so developers are aware of what exactly is stored.  I think we could refactor this to be consistent, but the danger right now is low as long as people use the usual access pattern.